internal/directives/git_pr_opener.go

```diff
@@ -198,7 +220,49 @@ func (g *gitPROpener) runPromotionStep(
 	}, nil
 }
 
-func getSourceBranch(sharedState State, cfg GitOpenPRConfig) (string, error) {
+// getPRNumber checks shared state for output from a previous execution of this
+// step. If any is found and it contains a PR number, that number is returned.
+// 0 is returned if no PR number is found in the shared state. An error is
```

Comment:
Nit: I think returning `-1` would be a more explicit "does not exist".

Reply:
I waffled on that myself. Since you're throwing your weight behind that option, I'll happily change it.

Reply:
Fixed in 0a593fd914852c4a26ee00cb02ed916624dbedeb